web-ui/src/main/resources/catalog/components/edit/onlinesrc/OnlineSrcDirective.js

```diff
@@ -1145,16 +1143,21 @@
                   scope.loadCurrentLink = function(url) {
                     scope.alertMsg = null;
 
-                    return gnOwsCapabilities.getWMSCapabilities(url)
-                        .then(function(capabilities) {
-                          scope.layers = [];
-                          scope.srcParams.selectedLayers = [];
-                          scope.layers.push(capabilities.Layer[0]);
-                          angular.forEach(scope.layers[0].Layer, function(l) {
-                            scope.layers.push(l);
-                            // TODO: We may have more than one level
+                    var serviceType = scope.srcParams.protocol.toLowerCase();
+                    if (serviceType.indexOf('ogc') !== -1) {
+                      return gnOwsCapabilities[
+                        serviceType.indexOf('wfs') !== -1 ?
+                          'getWFSCapabilities' : 'getWMSCapabilities'](url)
+                          .then(function(capabilities) {
+                            scope.layers = [];
+                            scope.srcParams.selectedLayers = [];
+                            scope.layers.push(capabilities.Layer[0]);
```

Comment:
Not an expert in WFS, but with some I've tested (at least version 1.1.0) there's no `Layer`, but `FeatureTypeList`with `FeatureType` entries, like:

```
<wfs:FeatureTypeList>
  <wfs:Operations>
    <wfs:Operation>Query</wfs:Operation>
  </wfs:Operations>
  <wfs:FeatureType xmlns:ps-f="urn:x-inspire:specification:gmlas:ProtectedSitesFull:3.0">
    <wfs:Name>ps-f:ProtectedSite</wfs:Name>
    <wfs:Title>ps-f:ProtectedSite</wfs:Title>
```